### PR TITLE
feat: Added log format support in syslog plugin.

### LINKED ISSUE #8278: feat: Support for custom log format in syslog plugin
### Description

Syslog plugin needs to support custom log formats.

### PR DESCRIPTION
**Pull request by fatihusta**
Fixes #8278

### CODE REVIEW

File: t/plugin/syslog.t [hunk 1]
```diff
@@ -132,8 +132,36 @@ passed
 [error]
 
 
+=== TEST 5: add plugin metadata
```
Comment (spacewander) [inline t/plugin/syslog.t:135]: Could you check the logged data format?
You can refer to https://github.com/apache/apisix/pull/7328/files#diff-1108dce1e7823ca0b746c4ba69e692e4b02c6d49117d70859954b56cbb810c29 as an example.

BTW, new test should be added at the end.

Comment (fatihusta) [reply t/plugin/syslog.t]: I fixed test reordering problem.

Should I add "access request" test?
And is that enough if is require?
```
=== TEST 10: access for log format
--- request
GET /hello
--- response_body
hello world
```

Comment (spacewander) [reply t/plugin/syslog.t]: Let's do it. And we need to inject some code in `extra_init_by_lua` to check the logged data.

Comment (fatihusta) [reply t/plugin/syslog.t]: I think we don't need `extra_init_by_lua` for these tests.
I added new test cases. Please review it again.

BTW: It was to hard to understand test framework for me. :)

Comment (spacewander) [reply t/plugin/syslog.t]: > BTW: It was to hard to understand test framework for me. :)

You can read the doc in https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md

File: t/plugin/syslog.t [hunk 2]
```diff
@@ -338,3 +338,83 @@ qr/sending a batch logs to 127.0.0.1:(\d+)/
...
+GET /hello
+--- response_body
+hello world
+--- wait: 0.5
+--- no_error_log
+[error]
+--- grep_error_log eval
+qr/sending a batch logs to 127.0.0.1:(\d+)/
+--- grep_error_log_out
+sending a batch logs to 127.0.0.1:5044
```
[the remark this answers is not in the document]
Comment (fatihusta) [reply t/plugin/syslog.t]: I changed syslog server port.

[the remark this answers is not in the document]
Comment (tzssangglass) [reply t/plugin/syslog.t]: Simply modifying the port is useless ， we should check the format of the reported logs to ensure that log_format is in effect, see: https://github.com/apache/apisix/blob/master/t/plugin/kafka-logger-log-format.t

[the remark this answers is not in the document]
Comment (fatihusta) [reply t/plugin/syslog.t]: Could you please check again?

File: docs/en/latest/plugins/syslog.md [hunk 3]
```diff
@@ -49,6 +49,40 @@ Logs can be set as JSON objects.
...
+:::info IMPORTANT
+
+Configuring the Plugin metadata is global in scope. This means that it will take effect on all Routes and Services which use the `syslog` Plugin.
+
+:::
+
+The example below shows how you can configure through the Admin API:
+
+```shell
+curl http://127.0.0.1:9080/apisix/admin/plugin_metadata/syslog -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```
Comment (spacewander) [inline docs/en/latest/plugins/syslog.md:69]:
```suggestion
curl http://127.0.0.1:9180/apisix/admin/plugin_metadata/syslog -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```

File: t/plugin/syslog.t [hunk 4]
```diff
@@ -324,3 +324,90 @@ qr/sending a batch logs to 127.0.0.1:(\d+)/
...
+    end
+--- request
+GET /hello
+--- response_body
+hello world
+--- wait: 0.5
+--- no_error_log
+[error]
+--- error_log eval
+qr/syslog-log-format.*\{.*"host":"localhost"/
```
[the remark this answers is not in the document]
Comment (fatihusta) [reply t/plugin/syslog.t]: I added upstream variable into log format.

**Comment by spacewander** [comment]
Please make the CI pass, thanks!

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**